pycqed/measurement/measurement_control.py

```diff
@@ -1673,26 +1744,85 @@ def save_timers(self, data_object, detector=True, MC=True):
                 log.error(f"Could not save timer for object: {obj}.")
                 traceback.print_exc()
 
-    def get_percdone(self):
-        percdone = self.total_nr_acquired_values / (
+    def get_percdone(self, current_acq=0):
+        """
+        Determine the current progress (in percent) based on how much data
+        MC has already received and possibly based on the progress of the
+        current acquisition.
+
+        In addition, get_percdone monitors whether progress has been made
+        since the last call to get_percdone, and it can log a warning to
+        slack and/or raise an exception if no progress is made for a
+        specified number of seconds (self.no_progress_interval and
+        self.no_progress_kill_interval, respectively).
+
+        :param current_acq: number of acquired samples in the current
+            acquisition. (Example: if 40 samples with averaging over 2**10
+            will be acquired in the current acquisition, and the current
+            averaging progress is 300, then current_acq should be set to
+            40*300/2**10.)
+        """
+        percdone = (self.total_nr_acquired_values + current_acq) / (
             np.shape(self.get_sweep_points())[0] * self.soft_avg()) * 100
+        try:
+            now = time.time()
+            if percdone != self._last_percdone_value:  # progress was made
+                self._last_percdone_value = percdone
+                self._last_percdone_change_time = now
+                log.debug(f'MC: percdone = {self._last_percdone_value} at '
+                          f'{self._last_percdone_change_time}')
+            elif self._last_percdone_change_time == 0:
+                # first progress check: initialize _last_percdone_change_time
+                self._last_percdone_change_time = now
+                self._last_percdone_log_time = self._last_percdone_change_time
+            else:  # no progress was made
+                no_prog_inter = getattr(self, 'no_progress_interval', 600)
+                no_prog_inter2 = getattr(self, 'no_progress_kill_interval',
+                                         np.inf)
```

Comment:
Would it make sense to make the above two into attributes as well at some point in init or run? So self.no_progress_interval and self.no_progress_kill_interval? Is there some reason?
Also where would they be created and by whom?

Reply:
I changed them to qcodes parameters now. I do not really understand the last question. No matter whether they are properties or parameters: they will be set by the user in the notebook and/or by the init script.

Reply:
Never mind, it was a stupid question. 
Thanks!